Misc/NEWS.d/next/Library/2023-05-01-18-53-20.gh-issue-102140._4gFLu.rst

```diff
@@ -0,0 +1,3 @@
+Enhanced the :meth:`has_header` in the :mod:`CSV` for better header detection.
+Now checks if all columns are strings and if the :py:data:`columnTypes` is empty.
+If conditions are met, it assigns the median length of the strings to :py:data:`columnTypes`.
```

Comment:
Same not about columnTypes here, plus I think the note goes into too much detail.